python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,22 +365,24 @@ def __get__(self, obj, cls):
             return cls._schema[name]
 
         if self._getter_name:
-            return self._bind_method(obj, self._getter_name)()
+            returnable = self._bind_method(obj, self._getter_name)()
+            if isinstance(returnable, GangaObject):
+                returnable._setParent(self)
+            return returnable
```

Comment:
I would have assumed that an object created in a getter method doesn't necessarily have to know about it's parent but I would have thought it nice to know the parent of such objects which aren't necessarily going through a setter method.
e.g. if I write a getter method to return a new Backend rather then always storing one (for whatever reason) I could imagine situations where you'd want to know which job object created this backend.
(The parent will never be set if the object is never assigned to the class within the getter)

I'm happy to revert if this causes issues but I was playing around with some classes and this cropped up as possible problem that I saw.